### PR TITLE
example: add FGVC-Aircraft dataset

### PR DESCRIPTION
**Pull request by anda-ren**
## Description
https://cloud.starwhale.cn/projects/14/datasets/6/versions/23/files

## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [x] Others

## Checklist
- [ ] run code format and lint check
- [ ] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: example/datasets/fgvc-aircraft/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  - `family`: String (such as 'Boeing 707')
+
+### Web viewer
+
+https://cloud.starwhale.cn/projects/14/datasets/6/versions/23/files
+
+## Use the `fgvc-aircraft-family-test` Dataset
+
+- Sign up with https://cloud.starwhale.cn
+```shell
```
Comment (tianweidut) [inline example/datasets/fgvc-aircraft/README.md:27]: add a blank line before line-27 ?

**Comment by tianweidut** [comment]
add user_raw format type for FGVC-Aircraft dataset? @anda-ren

File: example/datasets/fgvc-aircraft/README.md (hunk 1 above)
Comment (anda-ren) [reply example/datasets/fgvc-aircraft/README.md:28]: done

File: client/starwhale/base/bundle_copy.py [hunk 2]
```diff
@@ -202,10 +203,14 @@ def do(self) -> None:
             TransferSpeedColumn(),
             console=console,
             refresh_per_second=0.2,
-        ) as progress:
+        ) as progress, ThreadPoolExecutor(
+            max_workers=int(os.environ.get("SW_BUNDLE_COPY_THREAD_NUM", "5"))
+        ) as executor:
```
Comment (tianweidut) [inline client/starwhale/base/bundle_copy.py:208]: move 
```
ThreadPoolExecutor(
            max_workers=int(os.environ.get("SW_BUNDLE_COPY_THREAD_NUM", "5"))
        ) as executor
```
into `_do_upload_bundle_dir` function?

File: example/datasets/fgvc-aircraft/dataset.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    return _do_iter_raw_item("images_family_train.txt")
+
+
+def _do_iter_raw_item(fname):
+    with (DATA_DIR / fname).open("r") as f:
+        for line in f:
+            meta = line.split(" ", 1)  # ['1514522', 'Boeing 707']
+            annotations = {"family": meta[1].replace("\n", "")}
+            _img = DATA_DIR / "images" / f"{meta[0]}.jpg"
+            with PILImage.open(_img) as pil_image_open:
```
Comment (tianweidut) [inline example/datasets/fgvc-aircraft/dataset.py:52]:
```
 with PILImage.open(_img) as f:
                shape = f.size
```

the var for the context manager can use a short name.

Comment (anda-ren) [reply example/datasets/fgvc-aircraft/dataset.py]: done

File: client/starwhale/base/bundle_copy.py (hunk 2 above)
Comment (anda-ren) [reply client/starwhale/base/bundle_copy.py]: done

File: server/controller/src/main/java/ai/starwhale/mlops/datastore/ColumnTypeScalar.java [hunk 4]
```diff
@@ -161,12 +165,16 @@ public Object decode(Object value) {
...
+                return (short) (Integer.parseInt((String) value, 16) & 0xFFFF);
             } else if (this == INT32) {
-                return Integer.parseInt((String) value, 16);
+                return (int) (Long.parseLong((String) value, 16) & 0xFFFFFFFF);
             } else if (this == INT64) {
+                if (valueStr.length() == 16) {
+                    return (Long.parseLong(valueStr.substring(0, 8), 16) << 32) | Long.parseLong(
+                            valueStr.substring(8, 16), 16);
+                }
                 return Long.parseLong((String) value, 16);
```
[the remark this answers is not in the document]
Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/datastore/ColumnTypeScalar.java]: done

File: server/controller/src/main/java/ai/starwhale/mlops/datastore/ColumnTypeScalar.java [hunk 5]
```diff
@@ -161,12 +165,16 @@ public Object decode(Object value) {
...
                 throw new IllegalArgumentException("invalid bool value " + value);
             } else if (this == INT8) {
-                return Byte.parseByte((String) value, 16);
+                return (byte) (Integer.parseInt((String) value, 16) & 0xFF);
             } else if (this == INT16) {
-                return Short.parseShort((String) value, 16);
+                return (short) (Integer.parseInt((String) value, 16) & 0xFFFF);
             } else if (this == INT32) {
-                return Integer.parseInt((String) value, 16);
+                return (int) (Long.parseLong((String) value, 16) & 0xFFFFFFFF);
```
[the remark this answers is not in the document]
Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/datastore/ColumnTypeScalar.java]: done

**Review by xuchuan — CHANGES_REQUESTED** [review]
Please update encode and decode in client/starwhale/api/_impl/data_store.py to be consistent with the java implementation. See the implementation of float for reference.

**Comment by anda-ren** [comment]
> Please update encode and decode in client/starwhale/api/_impl/data_store.py to be consistent with the java implementation. See the implementation of float for reference.

done

**Review by xuchuan — APPROVED** [review]
LGTM

**Pull request merged**